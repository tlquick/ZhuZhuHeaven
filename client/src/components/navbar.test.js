import { BrowserRouter } from "react-router-dom";
import { render, screen, cleanup } from "@testing-library/react";
import renderer from "react-test-renderer";
import NavBar from "./navbar";
const mockedUseNavigate = jest.fn();

jest.mock("react-router-dom", () => ({
  ...jest.requireActual("react-router-dom"),
  useNavigate: () => mockedUseNavigate,
}));
jest.mock("./searchbar", () => () => {
  return <mock-search data-testid="search-1" />;
});
afterEach(() => {
  cleanup();
});
test("renders navbar component", () => {
  render(
    <BrowserRouter>
      <NavBar />
    </BrowserRouter>
  );
  const navbarElement = screen.getByTestId("navbar-1");
  expect(navbarElement).toBeInTheDocument();
  expect(navbarElement).toContainHTML("img"); //contains a <img> tag
  expect(navbarElement).toContainHTML("a"); //contains a <a> tag
});
test("matches navbar snapshot", () => {
  const tree = renderer.create(<NavBar />).toJSON();
  expect(tree).toMatchSnapshot();
});
