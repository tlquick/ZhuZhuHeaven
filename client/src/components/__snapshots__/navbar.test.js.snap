// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches navbar snapshot 1`] = `
<div
  className="header"
  data-testid="navbar-1"
>
  <a
    href="/"
  >
    <img
      alt="logo"
      src="/images/logo.png"
    />
  </a>
  <div
    className="nav"
  >
    <div>
      <p
        className="contact mx-auto fs-2"
      >
        Contact Us: 1800 zhu zhu
      </p>
      <p
        className="contact mx-auto fs-5"
      >
        ****Free Delivery in Australia
           
      </p>
    </div>
    <a
      className="link btn wiki fs-5"
      href="https://en.wikipedia.org/wiki/ZhuZhu_Pets"
    >
      Learn More
    </a>
    <a
      className="link btn"
      href="/user/signin"
    >
      <img
        alt="signin"
        src="/images/signin.png"
      />
    </a>
    <a
      className="link btn"
      href="/user/signup"
    >
      <img
        alt="signup"
        src="/images/signup.png"
      />
    </a>
    <button
      className="link btn"
      onClick={[Function]}
    >
      <img
        alt="signout"
        src="/images/signout.png"
      />
    </button>
    <a
      className="link btn"
      href="/cart"
    >
      <img
        alt="cart"
        src="/images/cart.png"
      />
    </a>
  </div>
  <img
    alt="pet logo"
    className="pet-logo"
    src="/images/zhuzhu.png"
  />
  <mock-search
    data-testid="search-1"
  />
</div>
`;
